### PR TITLE
Raise when attempting to write cell with perimeter data to SWC or ASC

### PR DESCRIPTION
**Pull request by wizmer**
and add missing binding for morphio::WriterError exception

Issue #126

### CODE REVIEW

File: tests/requirement_tests.txt [hunk 1]
```diff
@@ -1,3 +1,4 @@
 mock>=2.0.0
 nose>=1.3.7
 numpy>=1.14.2
+h5py>=2.9.0
```
Comment (tristan0x) [inline tests/requirement_tests.txt]: I would keep the file sorted.

File: tests/test_6_writers.py [hunk 2]
```diff
@@ -69,6 +72,10 @@ def test_write_basic():
         assert_array_equal(ImmutMorphology(os.path.join(tmp_folder, "test_write.swc")).points, expected)
         assert_array_equal(ImmutMorphology(os.path.join(tmp_folder, "test_write.h5")).points, expected)
 
+        with h5py.File(h5_out) as h5_file:
+            ok_(not '/perimeters' in h5_file.keys())
```
Comment (tristan0x) [inline tests/test_6_writers.py]: pep8 guideline
```suggestion
            ok_('/perimeters' not in h5_file.keys())
```

Besides, `.keys()` should not be required.
```suggestion
            ok_('/perimeters' not in h5_file)
```

File: src/mut/writers.cpp [hunk 3]
```diff
@@ -75,6 +82,9 @@ void swc(const Morphology& morphology, const std::string& filename)
         return;
     }
 
+    if (hasPerimeterData(morphology))
```
Comment (tristan0x) [inline src/mut/writers.cpp]: IMO it is more readable and less error-prone to put braces even around statements with one instruction.
Besides, there is a [clang-tidy check for that](https://clang.llvm.org/extra/clang-tidy/checks/readability-braces-around-statements.html) I recommend.

File: src/mut/writers.cpp [hunk 4]
```diff
@@ -171,6 +181,9 @@ void asc(const Morphology& morphology, const std::string& filename)
...
```
Comment (tristan0x) [inline src/mut/writers.cpp]: same remark about braces.

File: src/mut/writers.cpp [hunk 5]
```diff
@@ -319,7 +330,7 @@ void h5(const Morphology& morpho, const std::string& filename)
         // If the morphology has some perimeter data, we need to fill some
         // perimeter dummy value in the soma range of the data structure to keep
         // the length matching
-        if (hasPerimeterData)
+        if (hasPerimeterData_)
```
Comment (tristan0x) [inline src/mut/writers.cpp]: same remark about braces.

File: src/mut/writers.cpp [hunk 6]
```diff
@@ -368,7 +379,7 @@ void h5(const Morphology& morpho, const std::string& filename)
     write_attribute(h5_file, "comment",
         std::vector<std::string>{version_string()});
 
-    if (hasPerimeterData)
+    if (hasPerimeterData_)
```
Comment (tristan0x) [inline src/mut/writers.cpp]: same remark about braces.

File: src/mut/writers.cpp [hunk 7]
```diff
@@ -29,6 +29,13 @@ struct base_type<std::vector<T>> : base_type<T>
 {
 };
 
+bool hasPerimeterData(const morphio::mut::Morphology& morpho)
+{
+    return !morpho.rootSections().empty()
```
Comment (alkino) [inline src/mut/writers.cpp]: `return !morpho.rootSections().empty() &&!morpho.rootSections()[0]->perimeters().empty() ;`

seems me more common

File: src/mut/writers.cpp [hunk 8]
```diff
@@ -29,6 +29,13 @@ struct base_type<std::vector<T>> : base_type<T>
 {
 };
 
+bool hasPerimeterData(const morphio::mut::Morphology& morpho)
```
[the remark this answers is not in the document]
Comment (wizmer) [reply src/mut/writers.cpp]: I'm not a big fan of const on primitive types. But I tried and gcc gave me a warning:
`error: type qualifiers ignored on function return type [-Werror=ignored-qualifiers]`

File: src/errorMessages.cpp [hunk 9]
```diff
@@ -241,6 +241,11 @@ std::string ErrorMessages::ERROR_VECTOR_LENGTH_MISMATCH(const std::string& vec1,
     return msg;
 }
 
+std::string ErrorMessages::ERROR_PERIMETER_DATA_NOT_WRITABLE()
```
[the remark this answers is not in the document]
Comment (wizmer) [reply src/errorMessages.cpp]: I would have to change them all then. I will do in another commit.

File: src/mut/writers.cpp (hunk 7 above)
Comment (wizmer) [reply src/mut/writers.cpp]: Thanks.

File: src/mut/writers.cpp (hunk 8 above)
[the remark this answers is not in the document]
Comment (alkino) [reply src/mut/writers.cpp]: Hum

I talk of
bool hasPerimeterData(const morphio::mut::Morphology& morpho) const

[the remark this answers is not in the document]
Comment (wizmer) [reply src/mut/writers.cpp]: It's not a method in fact. Just a function.

[the remark this answers is not in the document]
Comment (alkino) [reply src/mut/writers.cpp]: oups sorry for that

**Pull request merged**